src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Socketer/SocketerClient.cs

```diff
@@ -553,6 +564,7 @@ private class SocketClient
             private bool threadShouldRun;
             // If a connection fails, wait this many ms
             private const int timeoutMS = 500;
+            private bool attemptConnection = true;
```

Comment:
nit: it might be better to make this and `shouldConnect` volatile if they are touched by multiple threads. It won't make a lot of difference here since they seem to be settable only once, but might avoid bugs down the line if the code changes.

Reply:
+1 for volatile. volatile also functions as a memory barrier in c# which is super useful.